.github/workflows/build-linux.yml

```diff
@@ -0,0 +1,178 @@
+
+      - name: Checkout ${{ github.event.inputs.commit }}
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.commit }}
+
+      - uses: dtolnay/rust-toolchain@stable
+
+      # TODO proper way to do cache?
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: ${{ runner.os }}-intel-cargo-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: 'TODO: Check if artifacts were already built and published before via .finished file'
+        run: echo "foo"
+
+      - name: Construct Build Command
+        id: construct_build_command
+        env:
+          TARGET_NAME: ${{ matrix.target.name }}
+          IMAGE: ${{ matrix.target.image }}
+          TARGET_STRING: ${{ matrix.target.target_string }} 
+          FEATURES_STRING: ${{ matrix.target.features_string }}
+        run: |
+          # full command
+          command="docker run \
+            -e SQLITE_MAX_VARIABLE_NUMBER=250000 \
+            -e SQLITE_MAX_EXPR_DEPTH=10000 \
+            -e LIBZ_SYS_STATIC=1 \
+            -w /root/build \
+            -v \"$(pwd)\":/root/build \
+            $IMAGE \
+            bash -c \
+              \" \
+              cargo clean \
+              && cargo build --release -p query-engine          --manifest-path query-engine/query-engine/Cargo.toml          $TARGET_STRING $FEATURES_STRING \
+              && cargo build --release -p query-engine-node-api --manifest-path query-engine/query-engine-node-api/Cargo.toml $TARGET_STRING $FEATURES_STRING \
+              && cargo build --release -p schema-engine-cli     --manifest-path schema-engine/cli/Cargo.toml                  $TARGET_STRING $FEATURES_STRING \
+              && cargo build --release -p prisma-fmt            --manifest-path prisma-fmt/Cargo.toml                         $TARGET_STRING $FEATURES_STRING \
+              \" \
+            "
+          # remove query-engine-node-api for "static" targets
+          if [[ "$TARGET_NAME" == *Static* ]]; then
+            substring_to_replace="&& cargo build --release -p query-engine-node-api --manifest-path query-engine/query-engine-node-api/Cargo.toml $TARGET_STRING $FEATURES_STRING"
+            replacement_string=""
+            command=$(echo "$command" | sed "s|$substring_to_replace|$replacement_string|")
+          fi
+          # store command in github output
+          echo "COMMAND=$command" >> "$GITHUB_OUTPUT"
+      - name: Show Build Command
+        env:
+          COMMAND: ${{ steps.construct_build_command.outputs.COMMAND }}"
+        run: echo "Build command is $COMMAND"
+      - name: Execute Build command
+        run:  ${{ steps.construct_build_command.outputs.command }}
+
+      - name: 'TODO: Compressing binary files'
+        run: echo "foo"
+      - name: 'TODO: do things with prismagraphql/build:release image (upload artifacts, verify uploaded artifacts)'
+        run: echo "foo"
```

Comment:
See https://github.com/prisma/engineer/blob/main/src/release/mod.rs#L127-L156
https://github.com/prisma/engineer/blob/main/static/validate.sh

- create the compressed archives
- create the .sig and sha files
- uploads the artifacts to s3 and r2
.gz
.gz.sig
.gz.sha256
.sha256
.sig